src/pages/SupplierRequisitionPage.js

```diff
@@ -379,45 +376,45 @@ export class SupplierRequisitionPage extends React.Component {
     const { isVaccine } = this.state;
     if (!isVaccine) return null;
 
-    const prevDate =
-      this.prevRequisititionDate === -Infinity ? new Date(2000, 1, 1) : this.prevRequisititionDate;
-    const { database } = this.props;
-    const dosesInVial = requisitionItem.item.doses;
+    const { database, requisition } = this.props;
 
-    const transactionBatchesCI = database
-      .objects('TransactionBatch')
-      .filtered(
-        'itemId = $0 && transaction.confirmDate >= $1 && transaction.type = $2',
-        requisitionItem.item.id,
-        prevDate,
-        'customer_invoice'
-      );
-    const doses = transactionBatchesCI.sum('doses');
+    // Get date of most recent requisition.
+
+    const prevRequisitions = database
+      .objects('Requisition')
+      .filtered('id != $0 && items.item.id = $1', requisition.id, requisitionItem.item.id)
+      .sorted('entryDate');
+
+    const isPrevRequisition = prevRequisitions.length > 0;
+
+    const prevRequisitionDate = isPrevRequisition ? prevRequisitions[0].entryDate : null;
+
+    // Get transactions for this item since most recent requisition.
 
-    const openVialWastage = transactionBatchesCI.sum('numberOfPacks') * dosesInVial - doses;
+    const transactionBatches = database.objects('TransactionBatch').filtered('itemId = $0', requisitionItem.item.id);
+          
+    const recentTransactionBatches = prevRequisitionDate ? transactionBatches.filtered('transaction.confirmDate >= $0', prevRequisitionDate) : transactionBatches;
```

Comment:
Quite a few of these lines are to long and this has been unformatted according to our rules!

Reply:
Whoops, I took prettier off `ctrl-s` because it kept fighting with the lint rules. Fixed by f8f8dfd1.